### PR TITLE
Improve message of discovery issues for ineffective `@Order` annotations

### LINKED ISSUE #4713: Improve "Ineffective @Order annotation" discovery issue by mentioning meta-annotations
The commit https://github.com/junit-team/junit-framework/commit/786de2932b4ae749af2753777cc93962f182ec12#diff-4b290a2104fbe0c8f688dfe991a5b5c668f62c6a561abb608421953fd8d5edfe introduced a warning about use of `@Order` without an orderer.

```
(1) [INFO] Ineffective @Order annotation on class 'com.example.OrderTest'. It will not be applied because ClassOrderer.OrderAnnotation is not in use.
    Source: ClassSource [className = 'com.example.OrderTest', filePosition = null]
            at com.example.OrderTest.<no-method>(SourceFile:0)
```

I do understand the problem, but the information does not help: in my case, a meta-annotation was annotated with `@Order` and **I had to use debug mode** to finally find this was due to a meta annotation.

The annotation was annotated like this:

```java
@Retention(RetentionPolicy.RUNTIME)
@Order(1)
public @interface ComposedAnnotation {
}
```

And the test:

```java
@ComposedAnnotation
public class OrderTest {

    @Test
    void test() {
        assertTrue(true);
    }
}
```

This result in the following warnings:

```
juil. 02, 2025 10:49:05 AM org.junit.platform.launcher.core.DiscoveryIssueNotifier logIssues
INFOS: TestEngine with ID 'junit-jupiter' encountered a non-critical issue during test discovery:

(1) [INFO] Ineffective @Order annotation on class 'com.example.OrderTest'. It will not be applied because ClassOrderer.OrderAnnotation is not in use.
    Source: ClassSource [className = 'com.example.OrderTest', filePosition = null]
            at com.example.OrderTest.<no-method>(SourceFile:0)
juil. 02, 2025 10:49:05 AM org.junit.platform.launcher.core.DiscoveryIssueNotifier logIssues
INFOS: TestEngine with ID 'junit-jupiter' encountered a non-critical issue during test discovery:

(1) [INFO] Ineffective @Order annotation on class 'com.example.OrderTest'. It will not be applied because ClassOrderer.OrderAnnotation is not in use.
    Source: ClassSource [className = 'com.example.OrderTest', filePosition = null]
            at com.example.OrderTest.<no-method>(SourceFile:0)
```

Could it be possible to add more details in those warnings?

## Context

 - Used versions (Jupiter/Vintage/Platform): Jupiter 5.13.2
 - Build Tool/IDE: Eclipse / Maven

### PR DESCRIPTION
**Pull request by marcphilipp**
## Overview

Resolves #4713.

---

I hereby agree to the terms of the [JUnit Contributor License Agreement](https://github.com/junit-team/junit-framework/blob/002a0052926ddee57cf90580fa49bc37e5a72427/CONTRIBUTING.md#junit-contributor-license-agreement).

---

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Method [preconditions](https://docs.junit.org/snapshot/api/org.junit.platform.commons/org/junit/platform/commons/util/Preconditions.html) are checked and documented in the method's Javadoc
- [x] [Coding conventions](https://github.com/junit-team/junit-framework/blob/HEAD/CONTRIBUTING.md#coding-conventions) (e.g. for logging) have been followed
- [x] Change is covered by [automated tests](https://github.com/junit-team/junit-framework/blob/HEAD/CONTRIBUTING.md#tests) including corner cases, errors, and exception handling
- [x] Public API has [Javadoc](https://github.com/junit-team/junit-framework/blob/HEAD/CONTRIBUTING.md#javadoc) and [`@API` annotations](https://apiguardian-team.github.io/apiguardian/docs/current/api/org/apiguardian/api/API.html)
- [x] Change is documented in the [User Guide](https://docs.junit.org/snapshot/user-guide/) and [Release Notes](https://docs.junit.org/snapshot/user-guide/#release-notes)

### CODE REVIEW

File: junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/discovery/ClassOrderingVisitor.java [hunk 1]
```diff
@@ -49,10 +48,12 @@ class ClassOrderingVisitor extends AbstractOrderingVisitor {
 		this.globalOrderer = createGlobalOrderer(configuration);
 		this.noOrderAnnotation = issueReporter.createReportingCondition(
 			testDescriptor -> !isAnnotated(testDescriptor.getTestClass(), Order.class), testDescriptor -> {
-				String message = "Ineffective @Order annotation on class '%s'. It will not be applied because ClassOrderer.OrderAnnotation is not in use.".formatted(
-					testDescriptor.getTestClass().getName());
+				String message = "Ineffective @Order annotation on class '%s'. ".formatted(
+					testDescriptor.getTestClass().getName())
+						+ "The annotation may be directly present or meta-present on the class. "
+						+ "It will not be applied because ClassOrderer.OrderAnnotation is not in use.";
```
Comment (sbrannen) [inline junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/discovery/ClassOrderingVisitor.java]:
```suggestion
				String message = """
						Ineffective @Order annotation on class '%s'. \
						It will not be applied because ClassOrderer.OrderAnnotation is not in use. \
						Note that the annotation may be either directly present or meta-present on the class."""//
						.formatted(testDescriptor.getTestClass().getName());
```

I prefer to use text blocks for such use cases.

**Review by sbrannen — CHANGES_REQUESTED** [review]
Looks good.

I only suggested the use of text blocks.

File: junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/discovery/MethodOrderingVisitor.java [hunk 2]
```diff
@@ -50,10 +49,12 @@ class MethodOrderingVisitor extends AbstractOrderingVisitor {
 		this.configuration = configuration;
 		this.noOrderAnnotation = issueReporter.createReportingCondition(
 			testDescriptor -> !isAnnotated(testDescriptor.getTestMethod(), Order.class), testDescriptor -> {
-				String message = "Ineffective @Order annotation on method '%s'. It will not be applied because MethodOrderer.OrderAnnotation is not in use.".formatted(
-					testDescriptor.getTestMethod().toGenericString());
+				String message = "Ineffective @Order annotation on method '%s'. ".formatted(
+					testDescriptor.getTestMethod().toGenericString())
+						+ "The annotation may be directly present or meta-present on the method. "
+						+ "It will not be applied because MethodOrderer.OrderAnnotation is not in use.";
```
Comment (sbrannen) [inline junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/discovery/MethodOrderingVisitor.java]:
```suggestion
				String message = """
						Ineffective @Order annotation on method '%s'. \
						It will not be applied because MethodOrderer.OrderAnnotation is not in use. \
						Note that the annotation may be either directly present or meta-present on the method."""//
						.formatted(testDescriptor.getTestMethod().toGenericString());
```

I would use a text block here, too.

Please note that I reordered the last two sentences and slightly modified the wording.

File: junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/discovery/ClassOrderingVisitor.java (hunk 1 above)
Comment (marcphilipp) [reply junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/discovery/ClassOrderingVisitor.java]: > I prefer to use text blocks for such use cases.

This "pattern" is so hard-wired into my brain that I didn't even think about using a text block. But I agree that it's more readable. :+1:

**Review by sbrannen — APPROVED** [review]
